docs/userguide/searchspace.md

```diff
@@ -1,46 +1,269 @@
+    NumericalDiscreteParameter(name="x0", values=[1, 2, 3]),
+    NumericalDiscreteParameter(name="x1", values=[4, 5, 6]),
+    CategoricalParameter(name="Speed", values=["slow", "normal", "fast"]),
+]
+subspace = SubspaceDiscrete.from_product(parameters=parameters)
+```
+
+In this example, `subspace` has a total of 27 different parameter configuration.
+
+~~~
+      x0   x1   Speed
+ 0   1.0  4.0    slow
+ 1   1.0  4.0  normal
+ 2   1.0  4.0    fast
+ ..  ...  ...     ...
+ 24  3.0  6.0    slow
+ 25  3.0  6.0  normal
+ 26  3.0  6.0    fast
+  
+  [27 rows x 3 columns]
+~~~
+
+### Constructing from a dataframe
+
+[`SubspaceDiscrete.from_dataframe`](baybe.searchspace.discrete.SubspaceDiscrete.from_dataframe) constructs a discrete subspace from a given dataframe.
+By default, this method tries to infer the data column as as a [`NumericalDiscreteParameter`](baybe.parameters.numerical.NumericalDiscreteParameter) and uses [`CategoricalParameter`](baybe.parameters.categorical.CategoricalParameter) as fallback.
+However, it is possible to change this behavior by using the optional `parameters` keyword.
+This list informs `from_dataframe` about the parameters and the types of parameters that should be used.
+In particular, it is necessary to provide such a list if there are non-numerical parameters that should not be interpreted as categorical parameters.
+
+```python
+import pandas as pd
+
+df = pd.DataFrame(
+    {
+        "x0": [2, 3],
+        "x1": [5, 4],
+        "x2": [9, 7],
+    }
+)
+subspace = SubspaceDiscrete.from_dataframe(df)
+```
+
+~~~
+ Discrete Parameters
+   Name                        Type  Num_Values Encoding
+ 0   x0  NumericalDiscreteParameter           2     None
+ 1   x1  NumericalDiscreteParameter           2     None
+ 2   x2  NumericalDiscreteParameter           2     None
+~~~
+
+### Creating a Simplex-Bound `SubspaceDiscrete`
+
+[`SubspaceDiscrete.from_simplex`](baybe.searchspace.discrete.SubspaceDiscrete.from_simplex) can be used to efficiently create a discrete search space (or discrete subspace) that is restricted by a simplex constraint, limiting the maximum sum of the parameters per dimension.
+This method uses a shortcut that removes invalid candidates already during the creation of parameter combinations and avoids to first create the full product space before filtering it.
+
+In the following example, a naive construction of the subspace would first construct the full product space, containing 25 points, although only 15 points are actually part of the simplex.
+
+```python
+parameters = [
+    NumericalDiscreteParameter(name="p1", values=[0, 0.25, 0.5, 0.75, 1]),
+    NumericalDiscreteParameter(name="p2", values=[0, 0.25, 0.5, 0.75, 1]),
+]
+subspace = SubspaceDiscrete.from_simplex(max_sum=1.0, simplex_parameters=parameters)
+```
+
+~~~
+       p1    p2
+ 0   0.00  0.00
+ 1   0.00  0.25
+ 2   0.00  0.50
+ ..   ...   ...
+ 12  0.75  0.00
+ 13  0.75  0.25
+ 14  1.00  0.00
+ 
+ [15 rows x 2 columns]
+~~~
+
```

Comment:
at least a sentence mentioning that one aslo has the opportunity to have parameters independent of the simplex via `product_parameters`